tests/test_s3.py

```diff
@@ -1543,6 +1543,64 @@ def test_has_statement_policy(self):
         resources = p.run()
         self.assertEqual(len(resources), 1)
 
+    def test_bucket_replication_policy_remove(self):
+        self.patch(s3.S3, "executor_factory", MainThreadExecutor)
+        self.patch(
+            s3.SetBucketReplicationConfig, "executor_factory", MainThreadExecutor
+        )
+        self.patch(
+            s3, "S3_AUGMENT_TABLE", [("get_bucket_policy", "Policy", None, "Policy")]
+        )
+        session_factory = self.replay_flight_data("test_s3_replication_policy_remove")
+        bname = "repela"
+        p = self.load_policy(
+            {
+                "name": "s3-has-replica-policy",
+                "resource": "s3",
+                "filters": [
+                    {"Name": bname}
+                ],
+                "actions": [
+                    {
+                        "type": "set-bucket-replication",
+                        "state": "remove"
+                    }
+                ]
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
```

Comment:
this test should verify the behavior of the action, by fetching the bucket replication to asserts its gone, also its best practice to assert an actual attribute of the resource, else the policy could return [1] and it would pass.

Reply:
Agreed. These tests were bare minimum and not very robust.

Resolved in ba52c08